### ISSUE TITLE
Migrate bazel's crosstools to features

### ISSUE THREAD

**Issue by mhlopko** [bazelbuild/bazel#5187]
Migrate all bazel's crosstools to the new style - action_configs and features. All (except msvc crosstools) bazel crosstools use legacy crosstool flags that will be eventually removed. The new style is superior because if allow for more precise flag handling (e.g. C vs c++), and easier to understand ordering.

**Comment by mhlopko** [comment]
This is currently blocked by #4571 - we cannot use features before we expose new Skylark api to the C++ toolchain, since the old api doesn't work with features.

**Comment by mhlopko** [comment]
Duplicate of https://github.com/bazelbuild/bazel/issues/5883
